cmd/kubeadm/app/cmd/init.go

```diff
@@ -309,6 +309,23 @@ func (i *Init) Run(out io.Writer) error {
 
 	adminKubeConfigPath := filepath.Join(kubeConfigDir, kubeadmconstants.AdminKubeConfigFileName)
 
+	if !i.dryRun {
+		// read the kubeadm log file and write a new one if it doesn't exist
+		data, err := kubeadmutil.ReadKubeadmLogFile()
```

Comment:
Thanks for following my comment about `WriteKubeadmLogFile`. However, we still need to invoke `ReadKubeadmLogFile` during dry runs.

Usually, a user will use `--dry-run` in order to see if the command will be successful and what resources will generate. Current code may have a successful dry run and not successful actual run (due to previous init or join).

I would have also moved the read to a pre-flight check and invoke the write at the very end of the init and join processes.

Reply:
> Usually, a user will use --dry-run in order to see if the command will be successful and what resources will generate. Current code may have a successful dry run and not successful actual run (due to previous init or join).

if we apply the same logic here:
https://github.com/kubernetes/kubernetes/blob/202fb7168233c64a3e26da5e7a23b095f33b7622/cmd/kubeadm/app/cmd/init.go#L249

it will pass during a dry run but can fail during an actual run.

> I would have also moved the read to a pre-flight check and invoke the write at the very end of the init and join processes.

i went away from the preflight idea for a couple of reasons:
- this isn't exactly a preflight check because other preflight checks don't perform an action after something fails, so that it doesn't fail.
in this case we write a file if a file cannot be read.
- it complicates the contribution for the new contributor